specification/connector/_types/Connector.ts

```diff
@@ -0,0 +1,267 @@
+  tooltip?: string
+  type: ConnectorFieldType
+  ui_restrictions: string[]
+  validations: Validation[]
+  value: ScalarValue
+}
+
+export type ConnectorConfiguration = Dictionary<
+  string,
+  ConnectorConfigProperties
+>
+
+export interface ConnectorScheduling {
+  enabled: boolean
+  /** The interval is expressed using the crontab syntax */
+  interval: string
+}
+
+interface CustomSchedulingConfigurationOverrides {
+  max_crawl_depth?: integer
+  sitemap_discovery_disabled?: boolean
+  domain_allowlist?: string[]
+  sitemap_urls?: string[]
+  seed_urls?: string[]
+}
+
+interface CustomScheduling {
+  configuration_overrides: CustomSchedulingConfigurationOverrides
+  enabled: boolean
+  interval: string
+  last_synced?: string
+  name: string
+}
+
+export type ConnectorCustomScheduling = Dictionary<string, CustomScheduling>
+
+export enum ConnectorStatus {
+  created,
+  needs_configuration,
+  configured,
+  connected,
+  error
+}
+
+export enum SyncStatus {
+  canceling,
+  canceled,
+  completed,
+  error,
+  in_progress,
+  pending,
+  suspended
+}
+
+export interface IngestPipelineParams {
+  extract_binary_content: boolean
+  name: string
+  reduce_whitespace: boolean
+  run_ml_inference: boolean
+}
+
+enum FilteringPolicy {
+  exclude,
+  include
+}
+
+enum FilteringRuleRule {
+  contains,
+  ends_with,
+  equals,
+  regex,
+  starts_with,
+  greater_than = '>',
+  less_than = '<'
+}
+
+interface FilteringRule {
+  created_at: string
+  field: string
+  id: string
+  order: integer
+  policy: FilteringPolicy
+  rule: FilteringRuleRule
+  updated_at: string
+  value: string
+}
+
+interface FilteringValidation {
+  ids: string[]
+  messages: string[]
+}
+
+enum FilteringValidationState {
+  edited,
+  invalid,
+  valid
+}
+
+interface FilteringAdvancedSnippet {
+  created_at: string
+  updated_at: string
+  value: Dictionary<string, UserDefinedValue>
+}
+
+interface FilteringRulesValidation {
+  errors: FilteringValidation[]
+  state: FilteringValidationState
+}
+
+interface FilteringRules {
+  advanced_snippet: FilteringAdvancedSnippet
+  rules: FilteringRule[]
+  validation: FilteringRulesValidation
+}
+
+export interface FilteringConfig {
+  active: FilteringRules
+  domain: string
+  draft: FilteringRules
+}
+
+interface FeatureEnabled {
+  enabled: boolean
+}
+
+interface SyncRulesFeature {
+  advanced?: FeatureEnabled
+  basic?: FeatureEnabled
+}
+
+export interface ConnectorFeatures {
+  document_level_security?: FeatureEnabled
+  filtering_advanced_config?: boolean
+  filtering_rules?: boolean
+  incremental_sync?: FeatureEnabled
+  sync_rules?: SyncRulesFeature
+}
+
+export interface SchedulingConfiguration {
+  access_control?: ConnectorScheduling
+  full?: ConnectorScheduling
+  incremental?: ConnectorScheduling
+}
+
+export interface Connector {
+  api_key_id?: string
+  configuration: ConnectorConfiguration
+  custom_scheduling: ConnectorCustomScheduling
+  description?: string
+  error: WithNullValue<string>
```

Comment:
Yes, I do agree with you here. Since the `Connector` denotes a response type. In a response, a missing (optional) field does indicate that the field is in fact null. 

I will adapt this, but I the meaningful is needed for requests definitions to set certain fields to null.